osu.Game/Overlays/Mods/ModSection.cs

```diff
@@ -50,6 +50,19 @@ public IEnumerable<Mod> Mods
 
                 ButtonsContainer.Children = modContainers;
                 buttons = modContainers.OfType<ModButton>().ToArray();
+
+                var expand = value.Any();
+
+                if (expand)
+                {
+                    headerLabel.FadeIn(200);
+                    Show();
+                }
+                else
+                {
+                    headerLabel.FadeOut(200);
+                    Hide();
```

Comment:
This hide is instant, meaning you'll never see the `FadeOut`. Maybe just use `this.FadeOut` and remove the `headerLabel` transition in both cases?

This whole addition could be reduced to `this.FadeTo(value.Any() ? 1 : 0, 200);`

Reply:
This feedback *still applies*.

You are missing the fact that Hide() is doing an instant hide. The headerLabel cannot fade out as you have specified because it is hidden by the parent element. Please explain your thought process and why you are rejecting the proposal in my above comment.

Reply:
to be more clear:
`Show()` = `this.FadeIn(0);`
`Hide()` = `this.FadeOut(0);`

Reply:
Sorry i though `Show()` is not equal to `this.FadeIn(0)` before ,my mistake.

Reply:
https://streamable.com/almtk
And i found an other issue.
If added a hidden object in `FillFlowContainer` and re-show it, the effect of moving position is not in the right place.